rust/tokens_transfer/src/tokens_transfer_backend/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+use candid::{CandidType, Principal};
+use std::hash::Hash;
+
+use ic_cdk_macros::*;
+use ic_ledger_types::{
+    AccountIdentifier, BlockIndex, Memo, Subaccount, Tokens, DEFAULT_SUBACCOUNT,
+    MAINNET_LEDGER_CANISTER_ID,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, Hash)]
+pub struct TransferArgs {
+    amount: Tokens,
+    to_principal: Principal,
+    to_subaccount: Option<Subaccount>,
+}
+
+#[update]
+async fn transfer(args: TransferArgs) -> Result<BlockIndex, String> {
+    ic_cdk::println!(
+        "Transferring {} tokens to principal {} subaccount {:?}",
+        &args.amount,
+        &args.to_principal,
+        &args.to_subaccount
+    );
+    let to_subaccount = args.to_subaccount.unwrap_or(DEFAULT_SUBACCOUNT);
+    let transfer_args = ic_ledger_types::TransferArgs {
+        memo: Memo(0),
+        amount: args.amount,
+        fee: Tokens::from_e8s(10_000),
+        // The subaccount of the account identifier that will be used to withdraw tokens and send them
+        // to another account identifier. If set to None then the default subaccount will be used.
+        // See the [Ledger doc](https://internetcomputer.org/docs/current/developer-docs/integrations/ledger/#accounts).
+        from_subaccount: None,
+        to: AccountIdentifier::new(&args.to_principal, &to_subaccount),
+        created_at_time: None,
+    };
+    ic_ledger_types::transfer(MAINNET_LEDGER_CANISTER_ID, transfer_args)
```

Comment:
It seems that your updated version of the example doesn't support interacting with the local version of the ledger, and then it wouldn't make sense to setup the local ledger at all, or am I missing something?

Reply:
the local and mainnet canister IDs are the same, thus it's ok to use `MAINNET_LEDGER_CANISTER_ID` here